internal/server/allocate_cases.go

```diff
@@ -0,0 +1,134 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"golang.org/x/sync/errgroup"
+)
+
+type AllocateCasesClient interface {
+	AllocateCases(ctx sirius.Context, assigneeID int, allocations []sirius.CaseAllocation) error
+	Case(ctx sirius.Context, id int) (sirius.Case, error)
+	Teams(ctx sirius.Context) ([]sirius.Team, error)
+}
+
+type allocateCasesData struct {
+	XSRFToken string
+	Entities  []string
+	Success   bool
+	Error     sirius.ValidationError
+
+	Teams            []sirius.Team
+	AssignTo         string
+	AssigneeUserName string
+}
+
+func AllocateCases(client AllocateCasesClient, tmpl template.Template) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if err := r.ParseForm(); err != nil {
+			return err
+		}
+
+		var caseIDs []int
+		for _, id := range r.Form["id"] {
+			caseID, err := strconv.Atoi(id)
+			if err != nil {
+				return err
+			}
+			caseIDs = append(caseIDs, caseID)
+		}
+
+		if len(caseIDs) == 0 {
+			return errors.New("id missing")
```

Comment:
Since this is shown to users, it would be good to make it something clearer like "no cases selected".